src/libraries/System.Private.CoreLib/src/System/IO/Strategies/AppendFileStreamStrategy.cs

```diff
@@ -0,0 +1,193 @@
+
+        internal override bool IsClosed => _fileHandle.IsClosed;
+
+        internal override SafeFileHandle SafeFileHandle => _fileHandle;
+
+        public override ValueTask DisposeAsync()
+        {
+            if (_fileHandle != null && !_fileHandle.IsClosed)
+            {
+                _fileHandle.ThreadPoolBinding?.Dispose();
+                _fileHandle.Dispose();
+            }
+
+            return ValueTask.CompletedTask;
+        }
+
+        internal override void DisposeInternal(bool disposing) => Dispose(disposing);
+
+        protected override void Dispose(bool disposing)
+        {
+            if (disposing && _fileHandle != null && !_fileHandle.IsClosed)
+            {
+                _fileHandle.ThreadPoolBinding?.Dispose();
+                _fileHandle.Dispose();
+            }
+        }
+
+        public override void Flush() { }  // no buffering = nothing to flush
+
+        public override Task FlushAsync(CancellationToken cancellationToken) => Task.CompletedTask; // no buffering = nothing to flush
+
+        internal override void Flush(bool flushToDisk)
+        {
+            if (flushToDisk && CanWrite)
+            {
+                FileStreamHelpers.FlushToDisk(_fileHandle);
+            }
+        }
+
+        public override long Seek(long offset, SeekOrigin origin)
+        {
+            if (origin < SeekOrigin.Begin || origin > SeekOrigin.End)
+            {
+                throw new ArgumentException(SR.Argument_InvalidSeekOrigin, nameof(origin));
+            }
+            else if (_fileHandle.IsClosed)
+            {
+                ThrowHelper.ThrowObjectDisposedException_FileClosed();
+            }
+            else if (!CanSeek)
+            {
+                ThrowHelper.ThrowNotSupportedException_UnseekableStream();
+            }
+
+            long length = Length;
+            long newLength = origin == SeekOrigin.Begin ? offset : length + offset;
+
+            if (newLength < 0)
+            {
+                // keep throwing the same exception we did when seek was causing actual offset change
+                FileStreamHelpers.ThrowInvalidArgument(_fileHandle);
+            }
+
+            // Prevent users from overwriting data in a file that was opened in append mode.
+            if (newLength < length)
+            {
+                throw new IOException(SR.IO_SeekAppendOverwrite);
+            }
+
+            FileStreamHelpers.SetFileLength(_fileHandle, newLength);
+
+            return newLength;
+        }
+
+        internal override void Lock(long position, long length) => FileStreamHelpers.Lock(_fileHandle, CanWrite, position, length);
+
+        internal override void Unlock(long position, long length) => FileStreamHelpers.Unlock(_fileHandle, position, length);
+
+        public override void SetLength(long value)
+        {
+            if (value < Length)
+            {
+                throw new IOException(SR.IO_SetLengthAppendTruncate);
+            }
```

Comment:
Similarly, this seems wrong.  Today this:
```C#
using System;
using System.IO;
using System.Text;

class Program
{
    static void Main()
    {
        const string Path = @"tmp.txt";
        File.Delete(Path);

        using (var fs = new FileStream(Path, FileMode.Append))
        {
            fs.Write(Encoding.ASCII.GetBytes("hello"));
            fs.SetLength(1);
        }

        Console.WriteLine(File.ReadAllText(Path));
    }
}
```
prints
```
h
```
but it seems like this code will end up throwing.